File: examples/python/agent_send_email.py
"""
Send an email (and reply) from an Inkbox agent identity.

Usage:
    INKBOX_API_KEY=ApiKey_... AGENT_HANDLE=sales-agent python agent_send_email.py
"""

import os
from inkbox import Inkbox

inkbox = Inkbox(api_key=os.environ["INKBOX_API_KEY"])
identity = inkbox.get_identity(os.environ["AGENT_HANDLE"])

# Agent sends outbound email
sent = identity.send_email(
    to=["recipient@example.com"],
    subject="Hello from your AI sales agent",
    body_text="Hi there! I'm your AI sales agent reaching out via Inkbox.",
    body_html="<p>Hi there! I'm your AI sales agent reaching out via <strong>Inkbox</strong>.</p>",
)
print(f"Sent message {sent.id}  subject={sent.subject!r}")

# Agent sends threaded reply
reply = identity.send_email(
    to=["recipient@example.com"],
    subject=f"Re: {sent.subject}",
    body_text="Following up as your AI sales agent.",
    in_reply_to_message_id=str(sent.id),
)
print(f"Sent reply {reply.id}")

File: examples/typescript/agent-send-email.ts
/**
 * Send an email (and reply) from an Inkbox agent identity.
 *
 * Usage:
 *   INKBOX_API_KEY=ApiKey_... AGENT_HANDLE=sales-agent npx ts-node agent-send-email.ts
 */

import { Inkbox } from "../../typescript/src/inkbox.js";

const inkbox = new Inkbox({ apiKey: process.env.INKBOX_API_KEY! });
const identity = await inkbox.getIdentity(process.env.AGENT_HANDLE!);

// Agent sends outbound email
const sent = await identity.sendEmail({
  to: ["recipient@example.com"],
  subject: "Hello from your AI sales agent",
  bodyText: "Hi there! I'm your AI sales agent reaching out via Inkbox.",
  bodyHtml: "<p>Hi there! I'm your AI sales agent reaching out via <strong>Inkbox</strong>.</p>",
});
console.log(`Sent message ${sent.id}  subject="${sent.subject}"`);

// Agent sends threaded reply
const reply = await identity.sendEmail({
  to: ["recipient@example.com"],
  subject: `Re: ${sent.subject}`,
  bodyText: "Following up as your AI sales agent.",
  inReplyToMessageId: sent.id,
});
console.log(`Sent reply ${reply.id}`);
